jujugui/static/gui/src/app/app.js

```diff
@@ -1254,12 +1185,26 @@ YUI.add('juju-gui', function(Y) {
         document.getElementById('header-search-container'));
     },
 
+    _renderModals: function() {
```

Comment:
missing docblock. 
Also maybe something a little less generic `_setupSettingsModals` ?
Then call this at the end of the init vs on every update (see below comment)

Why do we have to have these rendered at all time and then toggle their visibility? As it's a low traffic area I'd image we could get away with rendering/destroying when necessary for both of these modals.